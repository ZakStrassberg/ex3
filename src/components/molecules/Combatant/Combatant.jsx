import PropTypes from 'prop-types';
import React, { Component } from 'react';

import Checkbox from '../../atoms/Checkbox';

export default class MyComponent extends Component {
  static propTypes = {
    combatant: PropTypes.shape({
      id: PropTypes.number.isRequired,
      turnOver: PropTypes.bool.isRequired,
      initiative: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
    }).isRequired,
  };

  constructor(props) {
    super(props);

    this.state = {};
  }

  render() {
    const { combatant: { id, turnOver, initiative, name } } = this.props;
    return (
      <div key={id}>
        <Checkbox name={`turnover-${id}`} checked={turnOver} />
        - {initiative} - {name}
      </div>
    );
  }
}
