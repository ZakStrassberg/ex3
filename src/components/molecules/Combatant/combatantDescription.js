export const defaultCombatant = {
  id: 1,
  name: 'Harmonious Jade',
  initiative: 12,
  turnOver: false,
};
