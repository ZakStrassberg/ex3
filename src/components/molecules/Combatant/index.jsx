import Combatant from './Combatant';

export default Combatant;
