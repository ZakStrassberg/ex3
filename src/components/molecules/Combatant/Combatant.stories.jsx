import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs, text, boolean, number } from '@storybook/addon-knobs';

import Combatant from './';
import { defaultCombatant } from './combatantDescription';

const stories = storiesOf('Storybook Knobs', module);

// Add the `withKnobs` decorator to add knobs support to your stories.
// You can also configure `withKnobs` as a global decorator.
stories.addDecorator(withKnobs);

stories
  .add('Turn not over', () => <Combatant combatant={defaultCombatant} />)
  .add('Turn over', () => <Combatant combatant={{ ...defaultCombatant, turnOver: true }} />)
  .add('With knobs', () => {
    const combatantWithKnobs = {
      id: defaultCombatant.id,
      name: text('Name', defaultCombatant.name),
      turnOver: boolean('Turn Over', defaultCombatant.turnOver),
      initiative: number('Initiative', defaultCombatant.initiative),
    };

    return <Combatant combatant={combatantWithKnobs} />;
  });
