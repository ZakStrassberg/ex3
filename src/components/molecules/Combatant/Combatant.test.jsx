import React from 'react';

import { defaultCombatant } from './combatantDescription';
import Combatant from './';

describe('Checkbox', () => {
  it('should shallow render', () => {
    const wrapper = shallow(<Combatant combatant={defaultCombatant} />);
    expect(wrapper).toMatchSnapshot();
  });
});
