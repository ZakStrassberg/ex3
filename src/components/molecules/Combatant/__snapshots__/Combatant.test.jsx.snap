// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should shallow render 1`] = `
<div
  key="1"
>
  <Checkbox
    checked={false}
    label={null}
    name="turnover-1"
  />
  - 
  12
   - 
  Harmonious Jade
</div>
`;
